### ISSUE TITLE
eliminate 'sp' calls

### ISSUE THREAD

**Issue by dankelley** [dankelley/oce#2076]
See https://r-spatial.org/r/2023/04/10/evolution3.html#conserving-sp-workflows

**Comment by dankelley** [comment]
I think for `sp::point.in.polygon()` I can instead use `sf::intersects()`; see the start of the Example section at https://r-spatial.github.io/sf/reference/geos_binary_pred.html

**Comment by dankelley** [comment]
I just ran
```
_SP_EVOLUTION_STATUS_=2 R CMD check oce_1.8-0.tar.gz
```
and it had no problems, so I think that means that we don't have any actual problems, since whatever `sp` things I am using are not, in turn, using `rgdal`.  Even so, my impression is that `sf` is favoured over `sp`, so I will look into making a change.

**Comment by dankelley** [comment]
Below is a test code for `intersects()`.  (For some reason, reprex will not accept this code; it says that it crashes R.)
```R
library(sf)
pts <- st_sfc(st_point(c(.5,.5)), st_point(c(1.5, 1.5)), st_point(c(2.5, 2.5)))
pol <- st_polygon(list(rbind(c(0,0), c(2,0), c(2,2), c(0,2), c(0,0))))
lst <- st_intersects(pts, pol, sparse=FALSE)
plot(pts)
plot(pol, col=rgb(1,0,0,0.1), add=TRUE)
plot(pts[lst], col=2, pch=20, add=TRUE)
```

**Comment by dankelley** [comment]
Here is an idea of how kriging might be set up to use `sf` instead of `sp`.  I will check whether the results look OK in contours later (and will add to the present comment).
```R
library(oce)
library(sf)
library(automap)
data(wind, package="oce") # x y z
data <- st_as_sf(wind, coords=c("x", "y"))
x <- seq(0, 11, 1) # range(wind$x) # 0 10.8
y <- seq(0, 11, 1) # range(wind$y) # 0.3 8.3
grid <- st_as_sf(expand.grid(x=x, y=y), coords=c("x", "y"))
k <- autoKrige(z~1, data, grid)
plot(k)
```

**Comment by dankelley** [comment]
The `sp::point.in.polygon()` calls are a non-issue, according to the sp/sf experts.  So I'll ignore that.  I may still address the kriging issue, though, since I don't see any harm in doing so, and if oce code can prefer `sf::` to `sp::` throughout, then folks inspecting the code won't have to learn about two packages.

**Comment by dankelley** [comment]
I've updated "develop" so that
1. `mapPlot()` has deprecated the ability of `projection` to handle the output from `sp::CRS()`, because that function is now deprecated.
2. I changed the kriging method in `sectionSmooth()` to use `sf` objects, instead of `sp` objects.

Now, the only `sp` function in `oce` is in the use of `=sp::point.in.polygon()` within `map.R`.  I think that is okay, given what I've learned about `sp`.

PS. oh, and I also revived the documentation for the `method="kriging"` parameter of `sectionSmooth()`.  I had commented that out in the previous version because things were not working.  But I think things are okay now.  I'm attaching the 4-panel plot that is created by `examples(sectionSmooth)`.

[kriging2.pdf](https://github.com/dankelley/oce/files/11204615/kriging2.pdf)
